snakemake_executor_plugin_slurm/__init__.py

```
@@ -555,10 +555,15 @@ def test_account(self, account):
-                f"Unable to test the validity of the given or guessed SLURM account "
-                f"'{account}' with sacctmgr: {e.stderr}"
-            )
+            sacctmgr_report = f"Unable to test the validity of the given or guessed SLURM account '{account}' with sacctmgr: {e.stderr}."
+            try:
+                cmd = "sshare -U --format Account --noheader"
+                accounts = subprocess.check_output(
+                    cmd, shell=True, text=True, stderr=subprocess.PIPE
+                )
+            except subprocess.CalledProcessError as e:
+                sshare_report = "Unable to test the validity of the given or guessed SLURM account '{account}' with sshare: {e.stderr}."
+                raise WorkflowError(sacctmgr_report + " - and - " + sshare_report)
```

Ruff (0.8.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)